File: slack-poster/src/main/java/dev/number6/slackposter/dagger/SlackPosterConfigurationModule.java
package dev.number6.slackposter.dagger;

import dev.number6.slackposter.adaptor.EnvironmentVariableSlackPosterConfigurationAdaptor;
import dev.number6.slackposter.port.SlackPosterConfigurationPort;
import dagger.Module;
import dagger.Provides;

@Module
public class SlackPosterConfigurationModule {

    @Provides
    public SlackPosterConfigurationPort configPort() {
        return new EnvironmentVariableSlackPosterConfigurationAdaptor();
    }

}

File: slack-poster/src/main/java/dev/number6/slackposter/dagger/SlackPosterConfigurationModule.kt
package dev.number6.slackposter.dagger

import dagger.Module
import dagger.Provides
import dev.number6.slackposter.adaptor.EnvironmentVariableSlackPosterConfigurationAdaptor
import dev.number6.slackposter.port.SlackPosterConfigurationPort

@Module
class SlackPosterConfigurationModule {
    @Provides
    fun configPort(): SlackPosterConfigurationPort {
        return EnvironmentVariableSlackPosterConfigurationAdaptor()
    }
}